### ISSUE TITLE
Variable not highlighted correctly

### ISSUE THREAD

**Issue by leventelist** [microsoft/vscode-cpptools#9546]
Bug type: Language Service

**Describe the bug**
- OS and Version: Linux x64 5.10.0-15-amd64
- VS Code Version: Code 1.68.1 (https://github.com/microsoft/vscode/commit/30d9c6cd9483b2cc586687151bcbcd635f373630, 2022-06-15T02:58:26.441Z)
- C/C++ Extension Version: 1.11.0
- Other extensions you installed (and if the issue persists after disabling them):
- If using SSH remote, specify OS of remote machine:
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

**Steps to reproduce**

1. Load the sample code
2. Hover over timer at the parameter list
3. See error

**Expected behavior**
All variables named 'timer' shall be highlighted.

**Code sample and logs**

- Code sample

void simple_sched_start(simple_sched_timer_t *timer, simple_sched_jiffies_t delay_ms) {

simple_sched_jiffies_t local_jiffies;

ATOMIC_BLOCK(ATOMIC_FORCEON) {
local_jiffies = jiffies;
}

timer->start = local_jiffies; /This doesn't highlight/
timer->state = simple_sched_running;
timer->time = delay_ms / JIFFIES_MS;

}

- Configurations in `c_cpp_properties.json`

- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette

- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)

**Screenshots**

![](https://user-images.githubusercontent.com/14082763/177225564-668bbdbb-8fcd-41c5-86ed-8e6edd9af3e2.png)

**Additional context**

**Comment by leventelist** [comment]
[water.zip](https://github.com/microsoft/vscode-cpptools/files/9073357/water.zip)

Here is the whole project. It builds with some warnings.

**Comment by Colengms** [comment]
Hi @leventelist  .  I tried to repro the issue with the project you provided.  I extracted it on Ubuntu 20.04.  I already had avr-gcc installed and the Makefile Tools extension enabled.  I ran the `Makefile: Clean Configure` command, and it appeared to configure IntelliSense properly.  I clicked in the parameter name, and it correctly highlighted all instances of that variable.

![image](https://user-images.githubusercontent.com/49173979/178079629-05e2347f-a47b-41f7-af5d-61c05cb4a801.png)

So, I don't believe I'm able to reproduce the issue.  Is there some information I'm missing that I will need in order to repro?

I noticed that your image shows a squiggle within `ATOMIC_BLOCK(ATOMIC_FORCEON)`, and that line also does not appear to be colorized correctly.  A failure to correctly interpret use of that macro could explain a failure to interpret the variable on the subsequent line.  Are you sure you have IntelliSense configured properly, such that it's able to parse the content of all headers?  Perhaps you are not using Makefile Tools, which can configure IntelliSense based on your Makefile?
